### ISSUE TITLE
3d image regression

### ISSUE THREAD

**Issue by yenhochen** [NifTK/NiftyNet#73]
I'm trying to do a 3d image regression using the following configuration file. However, I keep getting nans during the training session. All of my images are standardized, cropped to the same dimension, and are given a generated mask. I assume the gradients are exploding. what can I do about this?

[INPUT]
spatial_window_size=(32,32,32)
csv_file=/work/radiology/s180052/TractTBI/test_in.csv
interp_order=3

[OUTPUT]
spatial_window_size=(32,32,32)
csv_file=/work/radiology/s180052/TractTBI/test_out.csv
interp_order=3

[MASK]
spatial_window_size=(32,32,32)
csv_file=/work/radiology/s180052/TractTBI/mask/mask.csv
interp_order=0

[TRAINING]
loss_type=RMSE
sample_per_volume=32
max_iter=500
save_every_n=100
max_checkpoints=5
optimiser=adam
lr=0.00000005
starting_iter=0
scaling_percentage=(-10.0, 10.0)
random_flipping_axes= 0,1

validation_every_n = 10
exclude_fraction_for_inference = 0.15
exclude_fraction_for_validation = 0.15

[NETWORK]
window_sampling=weighted
name=highres3dnet_small
activation_function=prelu
batch_size=2
volume_padding_size=(0, 0, 0)
whitening=True
normalisation=False

[SYSTEM]
cuda_devices=0
num_gpus=1
num_threads=4
queue_length=256
model_dir=/work/radiology/s180052/TractTBI/models

[REGRESSION]
image=INPUT
output=OUTPUT
sampler=MASK
loss_border=8
error_map=True

[INFERENCE]
border = (2,0,0)
output_interp_order = 3
save_seg_dir=/work/radiology/s180052/TractTBI/models/output
spatial_window_size = (8,256,256)

**Comment by wyli** [comment]
Hi @yenhochen a few things could cause NAN losses -- 1) 'nan' in the input 2) some image windows are "all zeros", the network might generate nans after a few layers of batch norm, 3)  the outputs' value range (see: https://stackoverflow.com/questions/48551744/sample-weighting-regression-niftynet)
